src/bin/radclient.c

```diff
@@ -1337,9 +1337,10 @@ int main(int argc, char **argv)
 		       break;
 
 		case 't':
-			if (!isdigit((int) *optarg))
-				usage();
-			timeout = atof(optarg);
+			if (fr_time_delta_from_str(&timeout, optarg, FR_TIME_RES_SEC) < 0) {
```

Comment:
That's fine.  Just to clear up the confusion.  Passing FR_TIME_RES_SEC, just makes the parsing function assume that unqualified values (i.e. those without s, us, ms, ns suffixes) are values in seconds.  The actual value is still provided in nanoseconds.